indico/modules/events/models/reviews.py

```diff
@@ -84,44 +103,90 @@ def get_reviewer_render_data(self, user):
                 'reviews': reviews}
 
 
+class ProposalMixin(object):
+    """Classes that represent a proposal object should extend this class (ex:
+    Abstract, Paper).
+    """
+    #: A unique identifier to handle rendering differences between proposal
+    #: types
+    proposal_type = None
+    #: Attribute to retrieve the object with access to the reviewing settings
+    call_for_proposals_attr = None
+    #: Whether there is support for multiple revisions per proposal or just one
+    revisions_enabled = True
+
+    # endpoints
+    delete_comment_endpoint = None
+    create_comment_endpoint = None
+    edit_comment_endpoint = None
+    create_review_endpoint = None
+    edit_review_endpoint = None
+
+    @property
+    def cfp(self):
+        return getattr(self.event_new, self.call_for_proposals_attr)
+
+    def get_revisions(self):
+        if self.revisions_enabled:
+            raise NotImplementedError
+        else:
+            return [self]
+
+    def get_last_revision(self):
+        if self.revisions_enabled:
+            raise NotImplementedError
+        else:
+            return self
+
+    def can_comment(self, user):
+        raise NotImplementedError
+
+    def can_review(self, user, check_state=False):
+        raise NotImplementedError
+
+    def get_delete_comment_url(self, comment):
+        return url_for(self.delete_comment_endpoint, comment)
+
+    def get_save_comment_url(self, comment=None):
+        return (url_for(self.edit_comment_endpoint, comment)
+                if comment
+                else url_for(self.create_comment_endpoint, self))
+
+    def get_save_review_url(self, group=None, review=None):
+        return (url_for(self.edit_review_endpoint, review)
+                if review
+                else url_for(self.create_review_endpoint, self, group))
+
+
 class ProposalCommentMixin(object):
     timeline_item_type = 'comment'
 
     def can_edit(self, user):
         raise NotImplementedError
 
 
-class ProposalCommentVisibility(RichIntEnum):
-    """Most to least restrictive visibility for abstract comments"""
-    __titles__ = [None,
-                  _("Visible only to judges"),
-                  _("Visible to conveners and judges"),
-                  _("Visible to reviewers, conveners, and judges"),
-                  _("Visible to contributors, reviewers, conveners, and judges"),
-                  _("Visible to all users")]
-    judges = 1
-    conveners = 2
-    reviewers = 3
-    contributors = 4
-    users = 5
-
-
 class ProposalReviewMixin(object):
+    """Classes that represent a review of a proposal should extend this class
```

Comment:
wrap after `"""` since it does not fit in one line. also, this doesn't look like a good "first" line of a docstring at all

I'd go for this instead:

```
"""Mixin for proposal reviews.

Classes that ...
"""
```